operators/endpointmetrics/controllers/observabilityendpoint/observabilityaddon_controller.go

```diff
@@ -226,6 +225,13 @@ func (r *ObservabilityAddonReconciler) initFinalization(
 		if err != nil {
 			return false, err
 		}
+
+		// revert the change to cluster monitoring stack
+		err = revertClusterMonitoringConfig(ctx, r.Client, installPrometheus)
```

Comment:
why move it out of `if !installPrometheus {`? Do we need to `revertClusterMonitoringConfig` for *KS?

Reply:
we don't need to create `cluster-monitoring-config` configmap for *KS, but we need to create the two secrets: `hub-alertmanager-router-ca` and `observability-alertmanager-accessor`. That's why I add this.